send.go

```diff
@@ -56,6 +60,65 @@ func (s *Sender) Send(flow *flow.Flow) {
 	s.agg.Add(flow)
 }
 
+// SendFlows sends the flows to the Kentik API, returning the number of bytes sent as the payload.
+func (s *Sender) SendFlows(flows []flow.Flow) (int64, error) {
+	if s.Device == nil {
+		return 0, fmt.Errorf("device not initialized")
+	}
+	if len(flows) == 0 {
+		return 0, nil
+	}
+
+	if s.Metrics != nil {
+		s.Metrics.TotalFlowsIn.Mark(int64(len(flows)))
+	}
+
+	// ensure all flows have the device ID set; otherwise it may not be properly queried
+	for i := range flows {
```

Comment:
Why this construct instead of `for _, flow := range flows`?

Reply:
Go. 

Using the `for _, flow := range flows` and then referencing/changing `flow` changes the copy but it does not get persisted in the backing slice/array unless we re-assign it back in. By using the index we modify the values within the slice itself, keeping any other values set on the object in the process.